modules/ballerina-config/src/main/java/org/ballerinalang/config/ConfigProcessor.java

```diff
@@ -0,0 +1,163 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.config;
+
+import org.ballerinalang.config.utils.parser.ConfigFileParser;
+import org.ballerinalang.config.utils.parser.ConfigParamParser;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * ConfigProcessor processes runtime, environment and config file configurations.
+ *
+ * @since 0.95
+ */
+public class ConfigProcessor {
+
+    private static final String BALLERINA_CONF = "ballerina.conf";
+    private static final String USER_DIR = "user.dir";
+    private static final String BALLERINA_CONF_DEFAULT_PATH = System.getProperty(USER_DIR) + File.separator +
+                                                                                                    BALLERINA_CONF;
+    private Map<String, String> runtimeParams = new HashMap<>();
+    private Map<String, String> resolvedGlobalConfigs = new HashMap<>();
+    private Map<String, Map<String, String>> resolvedInstanceConfigs = new HashMap<>();
+    private ConfigRegistry configRegistry;
+
+    public ConfigProcessor(ConfigRegistry configRegistry) {
+        this.configRegistry = configRegistry;
+    }
+
+    /**
+     * Sets runtime config properties gathered from user as a map.
+     *
+     * @param params The Ballerina runtime parameters (i.e: -B params)
+     */
+    public void setRuntimeConfiguration(Map<String, String> params) {
+        this.runtimeParams = params;
+    }
+
+    /**
+     * Processes runtime, environment and config file properties.This populates configRegistry with configs based on
+     * the following precedence order. 1. Ballerina runtime properties, 2. External config
+     * (environment vars, etcd or something similar), 3. ballerina.conf file
+     */
+    public void processConfiguration() throws IOException {
+        ConfigParamParser paramParser = new ConfigParamParser(runtimeParams);
+        Map<String, String> runtimeGlobalConfigs = paramParser.getGlobalConfigs();
+        Map<String, Map<String, String>> runtimeInstanceConfigs = paramParser.getInstanceConfigs();
+
+        File confFile = getConfigFile(runtimeParams.get(BALLERINA_CONF));
+
+        if (confFile != null) {
+            ConfigFileParser parser = new ConfigFileParser(confFile);
+            Map<String, String> fileGlobalConfigs = parser.getGlobalConfigs();
+            Map<String, Map<String, String>> fileInstanceConfigs = parser.getInstanceConfigs();
+
+            // TODO: make this variable replacement a pluggable process
+            // Give precedence to environment variables
+            lookUpVariables(fileGlobalConfigs, fileInstanceConfigs);
+
+            // Give precedence to system variables, if not overridden by environment variables
+//            lookUpSystemVariables(fileGlobalConfigs, fileInstanceConfigs);
```

Comment:
Remove the commented out line if not necessary

Reply:
Ack.